src/Sequence.cc

```diff
@@ -388,10 +388,17 @@ Sequence::execute_sequence_impl(Iterator step_begin, Iterator step_end, Context&
 {
     Iterator step = step_begin;
 
-    if (comm and comm->immediate_termination_requested_)
-        throw Error{ gul14::cat(abort_marker, "Stop on user request") };
     while (step < step_end)
     {
+        if (comm and comm->immediate_termination_requested_)
+        {
+            OptionalStepIndex idx = step - steps_.begin();
+            if (*idx == 0)
+                idx = gul14::nullopt;
+            else
+                idx = *idx - 1;
```

Comment:
In fact this is broken in loops I guess :-(

Reply:
Anyhow, I find it strange to mention an index that has not even started yet.
Opinions?

Reply:
... I think reporting the index of the step-that-we-were-about-to-start is perfectly fine. After all, this step just does not start because the user requested termination. If we did not have this check the termination would occur just *inside* this same step.

Reply:
Still itches me.

Maybe move the check into the Step (as you always wanted), and check it after the `Step started` message?